maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java

```diff
@@ -226,8 +232,21 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 					}
 					builder.updateModified(cpe.lastModified(), cpe.getPath());
 					buildpath.add(cpe);
+
+					if ((isWab && !hasWablibs) && !Artifact.SCOPE_PROVIDED.equals(artifact.getScope())
+						&& !Artifact.SCOPE_TEST.equals(artifact.getScope()) && !artifact.isOptional()) {
+
+						String fileNameMapping = MappingUtils
+							.evaluateFileNameMapping(MappingUtils.DEFAULT_FILE_NAME_MAPPING, artifact);
+						wablibs.add("WEB-INF/lib/" + fileNameMapping + "=" + cpe.getName() + ";lib:=true");
```

Comment:
Should `-wablib` be set to just the paths to the jar files? And Bnd will put them in WEB-INF/lib? For example, see https://github.com/bndtools/bnd/blob/61ed372bad81a7564dc1aa0195edd37502e8d601/biz.aQute.bndlib.tests/test/test/BuilderTest.java#L1628-L1650

Reply:
we can't, the names are different on the classpath than what the artifacts will be in the WEB-INF/lib, we need to use a resolver to do the mapping, just like I did.

Reply:
Well, it does not really matter what the names are inside the bundle :-) Just that they are there and on the BCP. But this is not important.

Reply:
sorry, what I mean to say is that the maven-jar-plugin will assign names to artifacts it places in the WAR, if we try to play with those names, we may end up with two jars of of the same thing in the WAR, which we don't want. So it's best to make sure we preserve these names and do the proper mapping.

Reply:
for instance, if you have a dependency which is locally built in the same project, artifact.getFile() will return a file who's name doesn't contain the version, BUT maven-war-plugin will shove a jar into the WEB-INF/lib/ that DOES have a version, which means if we also write a jar with no version into WEB-INF/lib then you'll have 

```
WEB-INF/lib/foo.jar
WEB-INF/lib/foo-3.4.6-SNAPSHOT.jar
```